plugins/linear-genome-view/src/LinearGenomeView/components/Highlight.tsx

```diff
@@ -0,0 +1,200 @@
+      label: 'Bookmark highlighted region',
+      icon: BookmarkIcon,
+      onClick: () => {
+        let bookmarkWidget = session.widgets.get('GridBookmark')
+        if (!bookmarkWidget) {
+          bookmarkWidget = session.addWidget(
+            'GridBookmarkWidget',
+            'GridBookmark',
+          )
+        }
+        // @ts-ignore
+        bookmarkWidget.addBookmark(model.highlight as Region)
+        dismissHighlight()
+      },
+    },
+  ]
+
+  function handleClose() {
+    setOpen(false)
+  }
+
+  if (!model.highlight) {
+    return
+  }
+
+  // coords
+  const mapCoords = (r: ParsedLocStringA) => {
+    const s = model.bpToPx({
+      refName: r.refName,
+      coord: r.start,
+    })
+    const e = model.bpToPx({
+      refName: r.refName,
+      coord: r.end,
+    })
+    return s && e
+      ? {
+          width: Math.max(Math.abs(e.offsetPx - s.offsetPx), 3),
+          left: Math.min(s.offsetPx, e.offsetPx) - model.offsetPx,
+        }
+      : undefined
+  }
+
+  const h = mapCoords(model.highlight as ParsedLocStringA)
+
+  return (
+    <>
+      {h ? (
+        <div
+          className={classes.highlight}
+          style={{
+            left: h.left,
+            width: h.width,
+            background: `${colord(color).alpha(0.35).toRgbString()}`,
+            borderLeft: `solid ${color}`,
+            borderRight: `solid ${color}`,
+          }}
+        >
+          <Tooltip title={'Highlighted from URL parameter'} arrow>
+            <IconButton ref={anchorEl} onClick={() => setOpen(true)}>
+              <LinkIcon
+                fontSize="small"
+                sx={{
+                  color: `${colord(color).darken(0.2).toRgbString()}`,
+                }}
+              />
+            </IconButton>
+          </Tooltip>
+          <Menu
+            anchorEl={anchorEl.current}
+            // anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+            onMenuItemClick={(_event, callback) => {
+              callback(session)
+              handleClose()
+            }}
+            open={open}
+            onClose={handleClose}
+            menuItems={menuItems}
+          />
+        </div>
+      ) : null}
+    </>
+  )
+})
+
+export const OverviewHighlight = observer(function OverviewHighlight({
+  model,
+  overview,
+}: {
+  model: LGV
+  overview: Base1DViewModel
+}) {
+  const { classes } = useStyles()
+  const color = useTheme().palette.quaternary?.main ?? 'goldenrod'
+
+  const { cytobandOffset } = model
+
+  // coords
+  const mapCoords = (r: ParsedLocStringA) => {
+    const s = overview.bpToPx({
+      ...r,
+      coord: r.reversed ? r.end : r.start,
+    })
+
+    const e = overview.bpToPx({
+      ...r,
+      coord: r.reversed ? r.start : r.end,
+    })
+
+    return s !== undefined && e != undefined
+      ? {
+          width: Math.abs(e - s),
+          left: s + cytobandOffset,
+        }
+      : undefined
+  }
+
+  const h = mapCoords(model.highlight as ParsedLocStringA)
```

Comment:
i believe this cast is "not valid", as model.highlight.start and model.highlight.end are potentially undefined by definition of ParsedLocString(?)

Reply:
Most recent commit: is making use of Required<ParsedLocString> https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype satisfactory?

Reply:
that is an interesting new feature of typescript, but I might opt for creating a new type (just the simple {assemblyName:string,refName:string,start:number,end:number} instead perhaps, and try to avoid using "as" cast entirely (casts are generally best if avoided as it cancels out any check that typescript would do for you)